CRM/Core/Permission.php

```diff
@@ -1187,6 +1187,14 @@ public static function getEntityActionPermissions() {
     ];
     $permissions['contribution_recur'] = $permissions['payment'];
 
+    $permissions['entity_batch']['create'] = $permissions['entity_batch']['delete'] = [
+      // This means that they can call create or delete if they have one of the
+      // below permissions. If they only have 'edit own' then the code *should* enforce
+      // that but the difference between the two seems to have
+      // been more of an intention to write code then something that
+      // was implemented....
+      ['edit own manual batches', 'edit all manual batches'],
+    ];
```

Comment:
I tried testing with a user that didn't have administer CiviCRM but has the batch permissions and I get a 403 error when trying to remove from batch. I haven't traced why yet.

Also thinking out loud if you had administer CiviCRM before the patch, but not one of these two, then after the patch the api would no longer allow you to create entity_batches. That's probably ok since it seems like an unlikely combo.

Reply:
yeah - I think we should probably implement the intent of the permissions

Reply:
@demeritcowboy where do you think we are with this? I'm not really vested in it so I'm OK to close it if it's unexpectedly tricky

Reply:
On the site where this came up permissions aren't an issue so I don't personally have a need for finer permissions.